### ISSUE TITLE
Commit and Status from the Git menu not functioning

### ISSUE THREAD

**Issue by rat-moonshine** [prominic/Moonshine-IDE#485]
While testing on Windows with latest Moonshine build, Commit and Status found opening blank window even if the project has some file modifications.

![image](https://user-images.githubusercontent.com/24370870/54590151-66d9d580-4a4d-11e9-9ec5-c0d554416721.png)

However, other actions from Git menu found working expectedly.

**Comment by rat-moonshine** [comment]
Fix tested working on Windows. Once I test on OSX also, I'll close this issue.

**Comment by rat-moonshine** [comment]
I have tested this working on OSX, too. I'm closing this issue.

**Comment by JoelProminic** [comment]
I noticed that the Commit window is still blank when I test it with a fresh local Git project:

1. Create a new project in Moonshine
2. Right click > Copy Path
3. Close project
4. Open project in CMD
5. Run "git init"
6. Reopen project in Moonshine
7. Git > Commit 

While the primary use case is a clone from GitHub, I think that we should at least investigate this logic to determine if this is a more fundemental issue.  I wouldn't expect the git status results to be that different for a new local Git project and a clone.

**Comment by rat-moonshine** [comment]
> I wouldn't expect the git status results to be that different for a new local Git project and a clone.

Yes, it is neither using any different execution in case of local Git project. While investigating on this, I noticed using Windows' shell-encoding to targeted diff-check file path works when a Git path exists in system environment. For some reason, the executed command never able to create the targeted file when no system-environment for Git is present (how this should be in fact) and using shell-encoding to its file path. 

I made some modification, and this should work now. Tested against local and cloud repositories on Windows and macOS.

**Comment by JoelProminic** [comment]
I tested this issue against my old Vagrant VM, and I got the same blank Commit window.

When I tested this issue on a more recent Windows version (https://app.vagrantup.com/StefanScherer/boxes/windows_10), the commit prompt populates like I expect.

However, I noticed these errors on the command line.  Is the Moonshine code trying to access these directories?

    : warning: could not open directory 'My Music/': Permission denied
    : warning: could not open directory 'My Pictures/': Permission denied
    : warning: could not open directory 'My Videos/': Permission denied

**Comment by rat-moonshine** [comment]
Having my Vagrant VM up, I able to reproduce the behavior in my tests as well. 

Having more tests, I noticed that the (Commit/Status) list does populates but it takes a bit longer than usual to get return from _NativeProcess_ in this case. I'm not sure what is happening inside the NativeProcess pipeline and application shell, but the Git command execution return is bit slow in this case, which can make the user feel its not working as immediate reaction; Due to non-existence of any UI notification.

Thus, I have added a busy indicator to the UI now, this will disappear once the process return successfully reached to the Commit/Status popup.

Please, give a test and wait for few seconds to ensure result return. I tested this on my Vagrant VM and after a few seconds' of wait the result successfully reached me.

**Comment by JoelProminic** [comment]
I ran a test with this on my Windows Vagrant VM.  The loading icon appeared properly, but the action did not complete, even though I let it run for 5+ minutes.

**Comment by JoelProminic** [comment]
On my Windows VDI, this seems to be working fine.   I see the progress icon for both the Commit and Status prompt.   I did notice that the prompt was very slow (5+ seconds for a tiny project) on some tests - do you have any ideas why there is a long delay like this?

**Comment by rat-moonshine** [comment]
No, not at this moment. I tested the command send to execute with NativeProcess immediately (after Commit action), but its return is unusually slow in this case. Maybe, I want to debug on this more, later.

**Comment by piotrzarzycki21** [comment]
@rat-moonshine please close that or move to next version

**Comment by rat-moonshine** [comment]
Since @JoelProminic still had trouble in his Vagrant tests, we would decide on this issue after discussing to him.

**Comment by JoelProminic** [comment]
Since this was still triggering for me in my last tests, let's keep this open for now for v2.1.0.

**Comment by JoelProminic** [comment]
I removed this from the 2.1.0 milestone, because there were no updates for this release, to my knowledge.   
I forgot about this issue and deleted the VM that had this problem.  If I can't reproduce it on a fresh VM, I will close this issue.

**Comment by JoelProminic** [comment]
I am still able to reproduce this error:
1. Create a new Windows Vagrant VM.  I used the "StefanScherer/windows_10" box in my test.
2. Install the latest Moonshine development build
3. Setup all applications with Moonshine SDK Installer
4. Clone the Moonshine repository:  https://github.com/prominic/Moonshine-IDE.git .   NOTE:  this download was very slow for me, which could be part of the problem.
5. Import MoonshineDESKTOPevolved project
6. Import MoonshineSharedCore project (May not be necessary)
7. Select MoonshineDESKTOPevolved and click Git > Status
8. BUG:  the Modified File(s) prompt appears, but the contents did not populate.  Eventually, I saw this error on the console:  "error: time thread did not terminate timely"
9. Open a command prompt, navigate to the project and try "C:\MoonshineSDKs\Git\git-2.20.1.\bin\git.exe status".  This returns almost immediately and reports no changes.
10. Add a file to MoonshineDESKTOPevolved
11. Select MoonshineDESKTOPevolved and click Git > Status.  This populates quickly and shows he added file
12. Delete the added file
13. Select MoonshineDESKTOPevolved and click Git > Status.  This populates quickly and shows no changes
12. Restart Moonshine
13. Select MoonshineDESKTOPevolved and click Git > Status.   This eventually completed, but it was very slow (20+ seconds)
